docs/specifications/tests/Zone-TP/zone11.md

```diff
@@ -0,0 +1,169 @@
+Z11_MULTIPLE_MNAME            | WARNING | ns_ip_list           | Distinct MNAMEs found, but only one is expected. Returned from name servers "{ns_ip_list}".
+Z11_MNAME_NOT_AUTHORITATIVE   | WARNING | ns                   | MNAME name server points to a non authoritative name server ("{ns}").
+Z11_MNAME_NO_RESPONSE         | NOTICE  | ns                   | MNAME name server points to a non responsive name server ("{ns}").
+Z11_MNAME_NOT_IN_DELEGATION   | INFO    | nsname               | MNAME name server {nsname} is not listed as NS record for the zone.
+Z11_MNAME_NOT_MASTER          | WARNING | ns_ip_list           | MNAME name server doesn't have the highest SERIAL. Returned from name servers "{ns_ip_list}".
+Z11_MNAME_IS_MASTER           | DEBUG   | ns_ip_list           | MNAME name server does appear to be master. Returned from name servers "{ns_ip_list}".
+
+The value in the Level column is the default severity level of the message. The
+severity level can be changed in the [Zonemaster-Engine profile]. Also see the
+[Severity Level Definitions] document.
+
+The argument names in the Arguments column lists the arguments used in the
+message. The argument names are defined in the [argument list].
+
+## Test procedure
+
+In this section and unless otherwise specified below, the terms "[DNS Query]"
+follow the specification for DNS queries as specified in [DNS Query and Response Defaults].
+The handling of the DNS responses on the DNS queries follow, unless otherwise specified below,
+what is specified for [DNS Response] in the same specification.
+
+1. Create a [DNS Query] with query type SOA and query name *Child Zone* ("SOA Query")
+
+2. Obtain the set of name server IP addresses using [Method4] and [Method5] ("Name Server IP").
+
+3. Create the following empty sets:
+   1. Name server IP address, MNAME, SERIAL ("MNAME Nameserver")
+   1. Name server IP address, SERIAL ("SERIAL Nameservers")
+   4. Name server IP address ("MNAME Not Master")
+   5. Name server IP address ("MNAME Is Master")
+
+4. For each name server IP in *Name Server IP* do:
+   1. Send *SOA Query* to the name server and collect the response.
+   2. Go to next name server IP if at least one of the following criteria is met:
+      1. There is no DNS response.
+      2. [RCODE Name] of the response is not "NoError".
+      3. The AA flag is not set in the response.
+      4. There is no SOA record with owner name matching the query.
+   3. From the DNS response, add the MNAME field to the *MNAME Nameserver* set.
+   4. From the DNS response, add the SERIAL field and name server IP address to the *SERIAL Nameservers* set.
+   5. Go to next name server.
+
+5. If the set *MNAME Nameserver* contains more than one element, then output *[Z11_MULTIPLE_MNAME]*
+   with the name server IP addresses from the set, and terminate these procedures.
+
+6. Send *SOA Query* to the name server in *MNAME Nameserver*.

```

Comment:
Here you have a hidden DNS lookup that I think should be explicit.
1. Then name might be in-bailiwick, and then we must query at least one of the name servers for the zone.
2. The name might not exist, NXDOMAIN.
3. There might not be any address record, NODATA.
4. There might be both IPv4 and IPv6 address for the name.
5. There might be more than one IPv4 or IPv6 address for the name.
6. The address or addresses might be not routable (e.g. private).
7. The address or addresses might not be reachable.

Reply:
Fixed